backend/cmd/rewrite/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"bytes"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/net/html"
+	"log"
+	"net/http"
+)
+
+func (rw *rewriter) getDefaultHtml() *html.Node {
+	req, _ := http.NewRequest("GET", rw.webEndpoint, nil)
+	res, _ := http.DefaultClient.Do(req)
+
+	doc, _ := html.Parse(res.Body)
+	return doc
+}
+
+func (rw *rewriter) getDefaultHtmlString() string {
+	b := &bytes.Buffer{}
+	_ = html.Render(b, rw.getDefaultHtml())
+
+	return b.String()
+}
+
+type meta struct {
+	Title         string
+	Description   string
+	OGTitle       string
+	OGDescription string
+	OGUrl         string
+	OGImage       string
+	Favicon       string
+}
+
+func addTag(node *html.Node, tag string, attr map[string]string) {
+	var attributes []html.Attribute
+	for key, val := range attr {
+		attributes = append(attributes, html.Attribute{
+			Key: key,
+			Val: val,
+		})
+	}
+	node.AppendChild(&html.Node{
+		Data: tag,
+		Type: html.ElementNode,
+		Attr: attributes,
+	})
+}
+
+func addMetaProperty(node *html.Node, meta map[string]string) {
+	addTag(node, "meta", meta)
+}
+
+func (rw *rewriter) writeMeta(meta meta) string {
+	doc := rw.getDefaultHtml()
+	var f func(*html.Node)
+	f = func(n *html.Node) {
```

Comment:
Why not just have this as a normal function? Feels a bit excessive to define it dynamically

Reply:
Didn't realise recursiveness worked there.